src/udisksstate.c

```diff
@@ -2246,18 +2186,20 @@ static GVariant *
 udisks_state_get (UDisksState           *state,
                   const gchar           *key,
                   const GVariantType    *type,
-                  GError               **error)
+                  gboolean              *ok)
 {
   gchar *path = NULL;
   GVariant *ret = NULL;
   gchar *contents = NULL;
   GError *local_error = NULL;
   gsize length = 0;
 
+  *ok = TRUE;
+
   g_return_val_if_fail (UDISKS_IS_STATE (state), NULL);
   g_return_val_if_fail (key != NULL, NULL);
   g_return_val_if_fail (g_variant_type_is_definite (type), NULL);
-  g_return_val_if_fail (error == NULL || *error == NULL, NULL);
+  g_return_val_if_fail (ok != NULL, NULL);
```

Comment:
The macro `g_return_val_if_fail` will return the last parameter `NULL` if the expression fails.  In this case we really want `ok` to not be `NULL`.  Thus if `ok` has a value, it's not equal to `NULL` and our check `ok != NULL` evaluates to `TRUE` which is not a fail and we keep going in the function.